apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PooledSession.cs

```diff
@@ -143,6 +181,26 @@ internal static PooledSession FromSessionName(SessionPool.ISessionPool pool, Ses
         /// </remarks>
         public void Dispose() => ReleaseToPool(false);
 
+        /// <summary>
+        /// Usually disposing of a <see cref="PooledSession"/> instance should release its
+        /// underlying session to the pool. But when we create a new instance of <see cref="PooledSession"/>
+        /// from an existing instance of <see cref="PooledSession"/>, that is, with the same underlying session,
+        /// the original instance shouldn't be allowed to be disposed of normally so that the underlying session
+        /// is not released back to the pool.
+        /// </summary>
+        /// <returns><code>true</code> if the <see cref="PooledSession"/> had not been disposed before.
+        /// <code>false</code> otherwise.</returns>
+        private bool MarkAsDisposed()
+        {
+            bool wasDisposed = Interlocked.Exchange(ref _disposed, 1) == 1;
+            if (!wasDisposed)
+            {
+                GC.SuppressFinalize(this);
```

Comment:
I don't think we even have a finalizer now - this may have been left over from a time that we did. Not suggesting a change right now, but we may be able to simplify this method to a single expression.